### PR TITLE
QE: Add client support for 15.6 client

### PR DESCRIPTION
**Pull request by vandabarata**
## What does this PR change?
- Adds client support for openSUSE Leap 15.6
- Adds client to the controller
- Updates mirror configuration

### CODE REVIEW

File: salt/mirror/etc/minima.yaml [hunk 1]
```diff
@@ -438,6 +438,10 @@ http:
     archs: [x86_64]
   - url: http://download.opensuse.org/update/leap/15.4/sle
     archs: [x86_64]
+  - url: http://download.opensuse.org/update/leap/15.5/sle
+    archs: [x86_64]
+  - url: http://download.opensuse.org/update/leap/15.6/sle
```
[the remark this answers is not in the document]
Comment (vandabarata) [reply salt/mirror/etc/minima.yaml:443]: it's there too, I just have no idea what belongs where in the mirror and have just been "copying" the ones before :sweat_smile: I can remove it if it doesn't belong here

[the remark this answers is not in the document]
Comment (vandabarata) [reply salt/mirror/etc/minima.yaml:443]: both of them

File: modules/controller/main.tf [hunk 2]
```diff
@@ -145,6 +145,8 @@ module "controller" {
     opensuse154arm_sshminion = length(var.opensuse154arm_sshminion_configuration["hostnames"]) > 0 ? var.opensuse154arm_sshminion_configuration["hostnames"][0] : null
     opensuse155arm_minion    = length(var.opensuse155arm_minion_configuration["hostnames"]) > 0 ? var.opensuse155arm_minion_configuration["hostnames"][0] : null
     opensuse155arm_sshminion = length(var.opensuse155arm_sshminion_configuration["hostnames"]) > 0 ? var.opensuse155arm_sshminion_configuration["hostnames"][0] : null
+    opensuse156arm_minion    = length(var.opensuse156arm_minion_configuration["hostnames"]) > 0 ? var.opensuse156arm_minion_configuration["hostnames"][0] : null
+    opensuse156arm_sshminion = length(var.opensuse156arm_sshminion_configuration["hostnames"]) > 0 ? var.opensuse156arm_sshminion_configuration["hostnames"][0] : null
```
Comment (srbarrios) [inline modules/controller/main.tf:149]: What about x86_64 flavours?

File: modules/controller/variables.tf [hunk 3]
```diff
@@ -687,6 +687,20 @@ variable "opensuse155arm_sshminion_configuration" {
...
+  }
+}
+
+variable "opensuse156arm_sshminion_configuration" {
+  description = "use module.<OPENSUSE156ARM_SSHMINION>.configuration, see main.tf.libvirt-testsuite.example"
+  default = {
+    hostnames = []
+  }
+}
+
```
Comment (srbarrios) [inline modules/controller/variables.tf:703]: Let's add the possibility to use x86_64 too.

File: salt/controller/bashrc [hunk 4]
```diff
@@ -102,6 +102,8 @@ export VIRTHOST_KVM_PASSWORD="linux" {% else %}# no KVM host defined {% endif %}
 {% if grains.get('opensuse154arm_sshminion') | default(false, true) %}export OPENSUSE154ARM_SSHMINION="{{ grains.get('opensuse154arm_sshminion') }}" {% else %}# no OPENSUSE154ARM ssh minion defined {% endif %}
 {% if grains.get('opensuse155arm_minion') | default(false, true) %}export OPENSUSE155ARM_MINION="{{ grains.get('opensuse155arm_minion') }}" {% else %}# no OPENSUSE155ARM minion defined {% endif %}
 {% if grains.get('opensuse155arm_sshminion') | default(false, true) %}export OPENSUSE155ARM_SSHMINION="{{ grains.get('opensuse155arm_sshminion') }}" {% else %}# no OPENSUSE155ARM ssh minion defined {% endif %}
+{% if grains.get('opensuse156arm_minion') | default(false, true) %}export OPENSUSE156ARM_MINION="{{ grains.get('opensuse156arm_minion') }}" {% else %}# no OPENSUSE156ARM minion defined {% endif %}
+{% if grains.get('opensuse156arm_sshminion') | default(false, true) %}export OPENSUSE156ARM_SSHMINION="{{ grains.get('opensuse156arm_sshminion') }}" {% else %}# no OPENSUSE156ARM ssh minion defined {% endif %}
```
Comment (srbarrios) [inline salt/controller/bashrc:106]: Let's add the possibility to use x86_64 too.

File: salt/mirror/etc/minima.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (srbarrios) [reply salt/mirror/etc/minima.yaml:443]: In theory, all what we have in minima mirror configuration here, it should be aligned with the content we have in our internal minima mirror for CI/BV.

**Comment by srbarrios** [comment]
I would clarify in the title that we only cover libvirt case here, keeping other backends out of the scope for now.

**Pull request merged**